### ISSUE TITLE
key not found in berkley db

### ISSUE THREAD

**Issue by paliwalg** [Consensys-Incorporated/quorum#122]
Quorum version : 1.1.0

I have a five node quorum setup.
Not sure why suddenly for a running application on one day, private transactions starts failing on one of the nodes. 
Looking into constellation log, error is coming up key not found in berkley db.

Any clue, how to debug\troubleshoot this issue ?

Thanks

**Comment by patrickmn** [comment]
Key not found means that the payload wasn't received by this node. It's a normal message when the node isn't party to the transaction (its public key is not in `privateFor`.)

Are you adding this constellation node's public key in `privateFor`, and is this node the only node that hosts this private/public keypair? (If you have more than one node using the same keypair, the first one to contact the network will be the destination, and any other node with the same keypair will generate this warning. In other words, that kind of setup isn't currently supported.)

**Comment by paliwalg** [comment]
Yes, I am trying to do private transactions specifying its public key in the privateFor and yes this is the only node hosting this private/public key. 
Also note that when I do private transaction with this node, I do specify the privateFor for one more node.
Scenario is, i am doing private transaction from Node 1 specifying the privateFor for Node 2 & Node 3.
 I see transaction in effect on Node 3 but not on Node 2.
Checking the logs on Node 2 display the key not found error message with the immediate time stamp.
In the error message it displays two keys one key keep on changing every time while the other key is public key of Node 2.

The same setup, same type of transactions was working from last few days, since yesterday only I am seeing this issue.
Let me know if you need more information.

Thanks

**Comment by patrickmn** [comment]
I've added the `db-list-keys.py` script that may help verify the existence (or non-existence) of transaction payloads that are sent. (This should be run when the Constellation node isn't running.)

`/path/to/constellation-repo/scripts/db-list-keys.py qdata/constellation1/payload.db`

Are you getting an error on the sending node when attempting to send a private transaction?

**Comment by karnawat83** [comment]
We are also facing the similar kind of issue. 
Let me describe the scenario in details:
Details:

Consider we have only Node 1 and Node 2.

A contract was previously deployed to Node 1 with privateFor for Node2. Transactions send to this smart contract on Node1 with privateFor to Node2 works fine. Also transactions send to same smart contract on Node2 with privateFor to Node 1 works. 

If the same contract is tried to be deployed again, from Node1 with privateFor Node2, it gets deployed on Node2, but not on Node1. If the same contract is tried to be deployed from Node2 with privateFor for Node1, it gets deployed on Node2, but fails again on Node1.

If any transaction internally creates an instance of the contract, it fails on Node1 but works fine on Node2. 

Basically on Node1, we are not able create a new address for any Smart Contracts. We confirmed many times that the constallation keys are correct.

We are getting following errors in constellation logs:
"Key Not found in BerkeleyDb payload.db

It will be really helpful if anybody can suggest something on this.

**Comment by patrickmn** [comment]
Please indicate if this is still an issue as of Quorum 2.0.0
